rclpy/rclpy/timer.py

```diff
@@ -80,3 +84,65 @@ def time_since_last_call(self):
     def time_until_next_call(self):
         with self.handle as capsule:
             return _rclpy.rclpy_time_until_next_call(capsule)
+
+
+class Rate:
+    """A utility for sleeping at a fixed rate."""
+
+    def __init__(self, timer: Timer, *, context=None):
+        # Rate is a wrapper around a timer
+        self._timer = timer
+        self._is_shutdown = False
+        self._is_destroyed = False
+
+        # This event is set to wake sleepers
+        self._event = threading.Event()
+        self._lock = threading.Lock()
+        self._num_sleepers = 0
+
+        # Set event when timer fires
+        self._timer.callback = self._event.set
+
+        # Set event when ROS is shutdown
+        context.on_shutdown(self._on_shutdown)
```

Comment:
What if `context` is `None`? Use default context?

Reply:
Removed default value of `None` in 9ef2e97. Context can't be `None` because it is created by a `Node`, which handles using the default context.
https://github.com/ros2/rclpy/blob/b5b4e36c362be044f146df434e73328340c8843f/rclpy/rclpy/node.py#L137